ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/EventsRange.scala

```diff
@@ -69,18 +71,23 @@ private[events] object EventsRange {
       range: EventsRange[Long],
       pageSize: Int,
   ): Connection => Vector[A] = connection => {
-    val minPageSize = 10 min pageSize max (pageSize / 10)
-    val guessedPageEnd = range.endInclusive min (range.startExclusive + pageSize)
-    val arithPage =
-      read(range copy (endInclusive = guessedPageEnd), None, Some(pageSize))(connection)
-    val found = arithPage.size
-    if (guessedPageEnd == range.endInclusive || found >= minPageSize)
-      arithPage
-    else
-      arithPage ++ read(
-        range copy (startExclusive = guessedPageEnd),
-        Some(minPageSize - found),
-        Some(minPageSize - found),
-      )(connection)
+
+    @tailrec
+    def loop(newBegin: Long, result: Vector[A]): Vector[A] = {
+      val guessedPageEnd = range.endInclusive min (newBegin + pageSize)
+      val arithPage =
+        read(
+          EventsRange(startExclusive = newBegin, endInclusive = guessedPageEnd),
+          None,
```

Comment:
We does not seem to use limit anymore, just fetch hints. Was this intentional?
If yes some code can be flushed this-regard.